core/services/relay/evm/mercury/transmitter.go

```diff
@@ -444,29 +489,67 @@ func (mt *mercuryTransmitter) latestReport(ctx context.Context, feedID [32]byte)
 	req := &pb.LatestReportRequest{
 		FeedId: feedID[:],
 	}
-	resp, err := mt.rpcClient.LatestReport(ctx, req)
-	if err != nil {
-		mt.lggr.Warnw("latestReport failed", "err", err)
-		return nil, pkgerrors.Wrap(err, "latestReport failed")
-	}
-	if resp == nil {
-		return nil, errors.New("latestReport expected non-nil response")
-	}
-	if resp.Error != "" {
-		err = errors.New(resp.Error)
-		mt.lggr.Warnw("latestReport failed; mercury server returned error", "err", err)
-		return nil, err
+
+	var reports []*pb.Report
```

Comment:
Access to this slice the goroutines below needs to be synchronised.

Reply:
It is synchronized with `mu`

Reply:
missed that, skimming :facepalm: